WordPress/wordpress-playground
# Add @wp-playground/wordpress package – a new home for WordPress builds

## Description

> [!IMPORTANT]
> Don't be scared by the _File Changed: 5,000+_ message at the top. This PR can be reviewed just fine.
> 
> Most of the updates refer to moved WordPress static assets. The actual relevant changes proposed in this PR are small configuration updates in about 50 files.

Moves WordPress data modules to a new dependency-free module `@wp-playground/wordpress`. Now every Playground module may import WordPress data modules and, e.g., use them in unit tests.

Before this PR, WordPress data modules lived in the `@wp-playground/remote` package. They couldn't be easily imported, and even if they could, that would often create a circular dependency problem.

Related to https://github.com/WordPress/wordpress-playground/issues/463

## Implementation details

* All WordPress bundling functions are updated to point to the newly created module.
* The `@wp-playground/remote` public directory is now set to `packages/playground/wordpress/public` as that's where all the WordPress files live. The `remote` package ships no public assets of its own, except for the .htaccess file which is now handled by a dedicated plugin.
* The new module also exports `RecommendedPHPVersion = '8.0'` to help avoid hardcoding a version number in unit tests shipped by other Playground packages.
* `@wp-playground/wordpress` has a `bundle-wordpress` nx task that bundles WordPress. It's deliberately not called `build` so that nx does not mistake it for a build dependency (which would trigger it on every `npm run dev`).
* Loading the data module from the filesystem is implemented in `@php-wasm/node` by implementing the `getPreloadedPackage` Emscripten handler. The handler runs a few safety checks and then reads the data module from the disk via `readFileSync`.
* A new unit test for the `setSiteOptions()` Blueprint step has been added to confirm that WordPress is actually correctly loaded in Node.js.

## Remaining tasks

- [x] Run the new `bundle` task to confirm it creates the correct WordPress files in the correct directories.
- [x] Add a unit test to the new module to explicitly test loading WordPress (instead of implicitly relying on the `setSiteOption` test)

## Testing instructions

Confirm the CI tests pass. E2E tests will confirm the playground.wordpress.net website still loads all the supported WordPress versions, and the unit tests will confirm that WordPress is correctly loaded in Node.js.
